ngraph/frontend/onnx_import/src/op/log_softmax.cpp

```diff
@@ -17,32 +17,95 @@
+                const auto reshape_pattern =
+                    default_opset::Constant::create(element::i64, Shape{2}, {0, 1});
+                const auto reshaped_max =
+                    std::make_shared<default_opset::Reshape>(max, reshape_pattern, true);
+
+                const auto data_minus_max =
+                    std::make_shared<default_opset::Subtract>(coerced_data, reshaped_max);
+
+                const auto result = std::make_shared<default_opset::LogSoftmax>(data_minus_max, 1);
+                if (data.get_partial_shape().is_static())
+                {
+                    return ngraph::builder::opset1::reshape(result, data.get_shape());
+                }
+                else
                 {
-                    OutputVector inputs{node.get_ng_inputs()};
-                    const auto data = inputs.at(0);
-                    const auto data_rank = data.get_partial_shape().rank();
+                    const auto data_shape = std::make_shared<default_opset::ShapeOf>(data);
+                    return std::make_shared<default_opset::Reshape>(result, data_shape, false);
+                }
+            }
+
+            OutputVector log_softmax(const Node& node, const int64_t DEFAULT_AXIS)
+            {
+                OutputVector inputs{node.get_ng_inputs()};
+                const auto data = inputs.at(0);
+                const auto data_rank = data.get_partial_shape().rank();
 
-                    const auto axis = node.get_attribute_value<int64_t>("axis", 1);
+                NGRAPH_CHECK(data_rank.is_static(),
+                             "ONNX Softmax data rank needs to be known (static)");
+
+                const auto axis = node.get_attribute_value<int64_t>("axis", DEFAULT_AXIS);
+
+                std::shared_ptr<ngraph::Node> result;
+                switch (data_rank.get_length())
+                {
+                case 0:
+                {
+                    result = default_opset::Constant::create(data.get_element_type(), Shape{}, {1});
+                    break;
+                }
+                case 1:
+                {
+                    // checks if the axis belongs to the allowed values set (-1 and 0 for 1D)
+                    ngraph::normalize_axis(node.get_description(), axis, data_rank);
+                    result = std::make_shared<default_opset::LogSoftmax>(data, 0);
+                    break;
+                }
+                default:
+                {
                     const auto normalized_axis =
                         ngraph::normalize_axis(node.get_description(), axis, data_rank);
 
-                    const auto softmax =
-                        std::make_shared<default_opset::Softmax>(data, normalized_axis);
-                    return {std::make_shared<default_opset::Log>(softmax)};
+                    result = onnx_logsoftmax(data, normalized_axis);
+                    break;
+                }
                 }
```

Comment:
Is there a problem with indents?

Reply:
No, this is ok. This is what clang-format produced and the formatting CI check didn't complain.